### ISSUE TITLE
Crash with Pipewire-Jack 0.3.33, here's a workaround!

### ISSUE THREAD

**Issue by praashie** [wineasio/wineasio#30]
Since upgrading from Pipewire ~0.3.31 to 0.3.33, WineASIO has started to crash immediately when trying to use it with Pipewire's Jack implementation instead of a native Jack server.

More information about the crash: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1495

### Workaround for low-latency Wine applications
Use Pipewire for handling your sound cards, and run a Jack2 server with `alsa` backend and using `pipewire` as the device.
```
jackd -d alsa -d pipewire -p $PERIOD_SIZE -r $SAMPLE_RATE
```

Pipewire is not able to communicate with native Jack very well, so individual clients can't directly connect between native Jack and Pipewire. You can run an individual Wine application in the native Jack server and connect other Pipewire/PW-Jack clients to the endpoints of the Jack server.

In my case, I'm using FL Studio. I'm not satisfied with using FL Studio ASIO because Wine's own PulseAudio/ALSA drivers use hidden buffers to increase stability, with the penalty of unacceptable (~60 ms) extra latency for playing virtual instruments.

**Comment by ghost** [comment]
you are a lifesaver.

**Comment by szszoke** [comment]
@praashie how did you make `jackd` work like that?

To me it fails with messages saying that jackd was unable to acquire the pipewire device.

WineASIO works when I stop `pipewire` and just use `jackd` but that is not feasible.

**Comment by tank-trax** [comment]
this does not work here

`jackd -d alsa -d pipewire -p 2 -r 48000`

```
[E][04694.333873] default      | [       control.c:   64 jackctl_server_create()] deprecated
[W][04694.334000] default      | [       control.c:   87 jackctl_server_create2()] not implemented 0x55af0cf66f50 0x55af0cf670c0 (nil)
[W][04694.334035] default      | [       control.c:  174 jackctl_server_get_parameters()] 0x55af0d713e80: not implemented
[W][04694.334062] default      | [       control.c:  162 jackctl_server_get_drivers_list()] 0x55af0d713e80: not implemented
[W][04694.334085] default      | [       control.c:  245 jackctl_driver_get_name()] 0x55af0d713ed0: not implemented
Unknown driver "alsa"
[W][04694.334110] default      | [       control.c:  114 jackctl_server_destroy()] 0x55af0d713e80: not implemented
```

**Comment by falkTX** [comment]
those calls are from jackdbus, which is not supported under pipewire.

**Comment by falkTX** [comment]
so likely the original poster has a jack2 build with jackd/classic mode on but disabled jackdbus.
you cannot have this under debian based distros

**Comment by szszoke** [comment]
@falkTX is that an option during compilation or a command line switch?

**Comment by falkTX** [comment]
A build option. Debian has both jackd classic and jackdbus enabled on its builds.

**Comment by szszoke** [comment]
You are right. I had jack2-dbus installed on one of my machines and there it didn't work. I tested on another machine where jack2-dbus wasn't installed and there it worked.

The interesting thing is that the first machine is still not working after I uninstalled jack2-dbus

**Comment by rolesyuk** [comment]
Steps to reproduce the bug with OSS software (PortAudio) with the record example paex_record.exe (playback with paex_saw.exe works well):

On Windows machine:

- Intall https://github.com/msys2/msys2-installer/releases/download/2022-01-28/msys2-x86_64-20220128.exe
- pacman -Syu --needed base-devel mingw-w64-x86_64-toolchain git unzip
- wget https://www.steinberg.net/asiosdk
- unzip asiosdk
- mv asiosdk_2.3.3_2019-06-14 /usr/local/asiosdk2
- git clone https://github.com/PortAudio/portaudio
- cd portaudio
- ./configure --with-winapi=asio
- make
- mkdir pa-test
- cp bin/.libs/paex_record.exe bin/.libs/pa_devs.exe bin/.libs/paex_saw.exe /mingw64/bin/libstdc++-6.dll /mingw64/bin/libgcc_s_seh-1.dll /mingw64/bin/libwinpthread-1.dll pa-test/
- zip -r pa-test.zip pa-test

On Linux machine:

- unzip pa-test.zip && cd pa-test
- WINEPREFIX=~/.wine-test wine64 regsvr32 wineasio.dll
- WINEPREFIX=~/.wine-test wine paex_record.exe 
	
Output:
```
=== Now recording!! Please speak into the microphone. ===
*** stack smashing detected ***: terminated
0108:err:seh:call_stack_handlers invalid frame 00007FF511B65710 (0000000000022000-0000000000220000)
0108:err:seh:NtRaiseException Exception frame is not in stack limits => unable to dispatch exception.
```

[pa-test.zip](https://github.com/wineasio/wineasio/files/8008831/pa-test.zip)

**Comment by tank-trax** [comment]
Not sure when the fix was made but WINEASIO works with Pipewire 0.3.51 using Wine-Staging 7.9

![image](https://user-images.githubusercontent.com/41337033/169676228-70aa7132-fe3a-4918-aac1-3ea2d2c4c037.png)

**Comment by RThomasHyde** [comment]
Sorry for the necro post. I couldn't get wineasio working with pipewire. This workaround _sort  of_ works for me, but with one big drawback: my USB audio interface disappears whenever I start jackd. Wineasio works with jackd running, but the only inputs & outputs available are my onboard audio. The USB interface reappears as soon as I stop jackd. I'm kind of stumped!
